packages/agent/src/errors.ts

```diff
@@ -4,4 +4,9 @@
  *
  * @todo https://github.com/dfinity/agent-js/issues/420
  */
-export class AgentError extends Error {}
+export class AgentError extends Error {
+  constructor(public readonly message: string) {
+    super(message);
+    Object.setPrototypeOf(this, AgentError.prototype);
```

Comment:
Why this? Isn't it the default behavior when using `new`?

Reply:
This was from an attempt to verify the error as an `instanceof` the custom error class, but it wasn't successful

Reply:
You can set a static property and an instance variables so that you can:

```
error.kind === AgentError.kind
```

Not as nice, but it's something... hehe